spotlight-idea-plugin/src/main/kotlin/com/fueledbycaffeine/spotlight/idea/SpotlightProjectService.kt

```diff
@@ -61,19 +68,35 @@ class SpotlightProjectService(
     }
   }
 
-  private suspend fun readAndEmit(spotlightFileType: SpotlightFileType) {
+  private suspend fun readAndEmit(
+    spotlightFileChangeType: SpotlightFileChangeType,
+    readRules: () -> SpotlightRules = { rulesList.read() }
+  ) {
     // VFS CHANGES are delivered on EDT+write; move read off EDT
     withContext(Dispatchers.IO) {
-      when (spotlightFileType) {
-        SpotlightFileType.IDE_PROJECTS -> {
+      when (spotlightFileChangeType) {
+        SpotlightFileChangeType.IDE_PROJECTS -> {
           val paths = ideProjectsList.read()
-          _ideProjects.emit(paths)
+          val rules = readRules()
+          val implicitRules = rules.implicitRules
+          val typeSafeInferenceLevel = rules.typeSafeAccessorInference ?: TypeSafeAccessorInference.DISABLED
+          val projectName = rules.projectName ?: project.name
+          val ruleSet = computeSpotlightRules(rootDir, projectName, implicitRules, typeSafeInferenceLevel) { allProjects.value }
+          val allPaths = BreadthFirstSearch.flatten(paths, ruleSet)
+          _ideProjects.emit(allPaths)
```

Comment:
BFS is only returning the transitives of `paths`, so those need to be added to `paths` for the complete set of included projects

Reply:
[75a8e09](https://github.com/joshfriend/spotlight/pull/46/commits/75a8e0926ecd4d7e8d2ed23de2ca36b45a79bc2b)